pycbc/derived_parameters.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (C) 2017  Collin Capano
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides a library of functions that calculate waveform parameters
+from other parameters. All exposed functions in this module's namespace return
+one parameter given a set of inputs.
+"""
+
+import copy
+import numpy
+import lal
+from pycbc import coordinates
+from pycbc.detector import Detector
+
+#
+# =============================================================================
+#
+#                           Helper functions
+#
+# =============================================================================
+#
+def _ensurearray(arg):
```

Comment:
It's only used in the `primary_` and `secondary_` functions. In those I need to do truth testing, which is easier to do if the input is guaranteed to be an array. Note that I wrap the output of those functions `_formatreturn` though, which should return a float if a float was passed in. When you say that if you pass in floats on the command line, you get an array back, which function are you referring to?

It occurs to me that I could just code up the `primary_` and `secondary_` functions using standard python boolean comparisons, then vectorize them. Do you prefer that?